packages/fhir-router/src/fhirrouter.ts

```diff
@@ -265,17 +263,43 @@ export class FhirRouter extends EventTarget {
     this.router.add(method, path, handler, { interaction: interaction ?? 'operation' });
   }
 
-  find(method: HttpMethod, path: string): RouteResult<FhirRouteHandler, FhirRouteMetadata> | undefined {
-    return this.router.find(method, path);
+  find(method: HttpMethod, url: string): RouteResult<FhirRouteHandler, FhirRouteMetadata> | undefined {
+    let path = url;
+    let query: Record<string, string | string[]> | undefined;
+
+    const queryIndex = url.indexOf('?');
+    if (queryIndex >= 0) {
+      query = parseQueryString(url);
+      path = url.substring(0, queryIndex);
+    }
+
+    const result = this.router.find(method, path);
+    if (result && query) {
+      result.query = query;
+    }
+    return result;
   }
 
   async handleRequest(req: FhirRequest, repo: FhirRepository): Promise<FhirResponse> {
-    const result = this.router.find(req.method, req.pathname);
+    const url = req.url;
+    // User-specified URL is parsed into component parts, which are populated back onto the request
+    if (req.pathname) {
+      throw new OperationOutcomeError(badRequest('FhirRequest must specify url instead of pathname'));
+    }
+    const result = this.find(req.method, url);
     if (!result) {
       return [notFound];
     }
-    const { handler, params } = result;
+    const { handler, params, query } = result;
+
+    // Populate request object with parsed URL components from router
     req.params = params;
+    if (query) {
+      req.pathname = url.slice(0, url.indexOf('?'));
```

Comment:
Could the logic of separating `pathname` from the query string be centalized? Similar versions of splitting `url` on  `url.indexOf('?')` happen both here and in `find`.